tools-local/Microsoft.ML.CodeAnalyzer/NoThisAnalyzer.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Microsoft.ML.CodeAnalyzer
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public sealed class NoThisAnalyzer : DiagnosticAnalyzer
```

Comment:
:bulb: Consider using one of the maintained alternatives instead of roll-your-own analyzer for this.

Reply:
Point me at one please. Same for all your identical comments.

Reply:
Replaced with [SX1101](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/SX1101.md).